cmake/CodeCoverage.cmake

```diff
@@ -15,7 +15,6 @@ if ( NOT DEFINED CODECOV_HTMLOUTPUTDIR )
 endif ( NOT DEFINED CODECOV_HTMLOUTPUTDIR )
 
 if ( CMAKE_COMPILER_IS_GNUCXX )
-    find_program( CODECOV_GCOV gcov )
     find_program( CODECOV_LCOV lcov )
```

Comment:
Neat! 
I think we can also drop lcov as it is used with gcov together.

Reply:
Ohh I haven't looked too closely at the changes that you have made,  I thought that coveralls is doing all of the gcov+lcov runs.

Reply:
`coveralls` can do `gcov` runs but it'll be a hassle to configure that path now. I found the `lcov` approach more flexible and `coveralls` can use its output. If at any point we want to have html output we can use the other target added `coverage_html`.